src/test/java/ru/lanwen/wiremock/ext/WiremockResolverUnitTest.java

```diff
@@ -0,0 +1,135 @@
+package ru.lanwen.wiremock.ext;
+
+import com.github.tomakehurst.wiremock.WireMockServer;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ExtensionContext.Namespace;
+import org.junit.jupiter.api.extension.ExtensionContext.Store;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+import org.mockito.InOrder;
+import ru.lanwen.wiremock.config.CustomizationContext;
+import ru.lanwen.wiremock.config.CustomizationContext.CustomizationContextBuilder;
+import ru.lanwen.wiremock.config.WiremockCustomizer;
+import ru.lanwen.wiremock.ext.WiremockResolver.Wiremock;
+
+import java.lang.reflect.Parameter;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertSame;
+import static org.junit.Assert.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+import static org.junit.jupiter.api.extension.ExtensionContext.Namespace.create;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.inOrder;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.verifyZeroInteractions;
+import static org.mockito.Mockito.when;
+import static ru.lanwen.wiremock.ext.WiremockResolver.WIREMOCK_PORT;
+
+/**
+ * @author SourcePond (Roland Hauser)
+ */
+public class WiremockResolverUnitTest {
+    private static final int EXPECTED_PORT = 8088;
+    private WiremockFactory wiremockFactory = mock(WiremockFactory.class);
+    private WireMockServer server = mock(WireMockServer.class);
+    private CustomizationContextBuilder customizationContextBuilder = mock(CustomizationContextBuilder.class);
+    private CustomizationContext customizationContext = mock(CustomizationContext.class);
+    private WiremockCustomizer customizer = mock(WiremockCustomizer.class);
+    private ExtensionContext extensionContext = mock(ExtensionContext.class);
+    private ParameterContext parameterContext = mock(ParameterContext.class);
+    private Namespace namespace = create(WiremockResolver.class);
+    private Store store = mock(Store.class);
+    private WiremockResolver resolver = new WiremockResolver(wiremockFactory);
+    private Wiremock mockedServer;
+    private Parameter serverParameter;
+
+    private void supportedMethod(@Wiremock WireMockServer server) {
+
+    }
+
+    private void unsupportedMethod(WireMockServer server) {
+
+    }
+
+    @BeforeEach
+    public void setup() throws Exception {
```

Comment:
I think its too much efforts to do the test - integration test covers the same in a much better way